marshal.go

```diff
@@ -395,8 +416,9 @@ func (x *GoSNMP) send(packetOut *SnmpPacket, wait bool) (result *SnmpPacket, err
 		}
 	}
 
-	// detect unknown engine id error and retransmit with updated engine id
-	if len(result.Variables) == 1 && result.Variables[0].Name == ".1.3.6.1.6.3.15.1.1.4.0" {
+	// v3: detect unknown engine id error and retransmit with updated engine id
+	if result.Version == Version3 && result.PDUType == Report && len(result.Variables) == 1 &&
+		result.Variables[0].Name == ".1.3.6.1.6.3.15.1.1.4.0" {
```

Comment:
What about the case of out-of-time-window above? Now you do 2 similar checks a different way.

I would have put this part inside the `if result.Version == Version3` block from line 398 above.

Reply:
yup, that was not optimal. Refactored things a bit to make the check consistent with ``sendOneRequest()``